editor/src/clj/editor/outline_view.clj

```diff
@@ -471,15 +496,20 @@
                        (proxy-super setGraphic nil)
                        (proxy-super setContextMenu nil)
                        (proxy-super setStyle nil))
-                     (let [{:keys [label icon link color outline-error? outline-overridden? outline-reference? outline-show-link? parent-reference? child-error? child-overridden? scene-visibility]} item
+                     (let [{:keys [label icon link color outline-error? outline-overridden? outline-reference? outline-show-link? parent-reference? child-error? child-overridden? scene-visibility hideable? node-outline-key-path]} item
                            icon (if outline-error? "icons/32/Icons_E_02_error.png" icon)
                            show-link? (and (some? link)
                                            (or outline-reference? outline-show-link?))
-                           label (if show-link? (format "%s - %s" label (resource/resource->proj-path link)) label)]
-                       (proxy-super setText label)
-                       (proxy-super setGraphic (icons/get-image-view icon 16))
+                           text (if show-link? (format "%s - %s" label (resource/resource->proj-path link)) label)
+                           hidden? (= :hidden scene-visibility)
+                           visibility-icon (if hidden? eye-icon-closed eye-icon-open)]
+                       (.setImage image-view-icon (icons/get-image icon))
+                       (.setText text-label text)
+                       (.setGraphic visibility-button visibility-icon)
+                       (proxy-super setGraphic pane)
```

Comment:
I initially tried switching to a `TreeTableView` component that seemed like a better fit, but that would require a lot of changes that might produce more problems down the line. I worked around the issue here by using `setGraphic` to render the icon+label+toggle.